src/odemis/gui/cont/tabs.py

```diff
@@ -3312,40 +3312,47 @@ def __init__(self, name, button, panel, main_frame, main_data):
         self._speccnt_stream = None
         if "fiber-align" in tab_data.align_mode.choices:
             # Need to pick the right/best component which receives light via the fiber
-            fbdet = None
+            self._photods = []
             fbaffects = main_data.fibaligner.affects.value
             # First try some known, good and reliable detectors
             for d in (main_data.spectrometer, main_data.cld):
                 if d is not None and d.name in fbaffects:
-                    fbdet = d
-                    break
-            else:
+                    self._photods.append(d)
+
+            # This can be deleted once the PR SPARC detector with role numbered is merged
```

Comment:
This PR is merged, so this could be "deleted". At least, this comment, as for the rest, I wonder if it could still worthy to keep this fallback handling. Now it'll look at every spectrometer and photo detectors. Admittedly, if there are components which are not any of these, there are also probably not worthy to be used.